services/src/panic.c

```diff
@@ -46,29 +55,48 @@ void panic_set_hook(const PanicHook panicHookFunction, void* reserved)
     panicHook = panicHookFunction;
 }
 
-void panic_(const ePanicCode code, void* extraInfo, void(*unnamed)(uint32_t))
+void panic_(const ePanicCode code, const char* text, void* unused) {
+    PANIC_COMPAT(code, text);
+}
+
+void panic_ext(const PanicData* data, void* reserved)
 {
     #if HAL_PLATFORM_CORE_ENTER_PANIC_MODE
         HAL_Core_Enter_Panic_Mode(NULL);
     #else
         HAL_disable_irq();
     #endif // HAL_PLATFORM_CORE_ENTER_PANIC_MODE
 
+    memset(&g_retainedPanicData, 0, sizeof(g_retainedPanicData));
+    g_retainedPanicData.marker = PANIC_DATA_MARKER;
+    memcpy(&g_retainedPanicData.data, data, sizeof(PanicData));
```

Comment:
kind of a nit: but do we want to only set the marker after the data has been copied? Swap the order of these two lines?

Reply:
:+1: Good idea.